library/common/main_interface.h

```diff
@@ -29,6 +29,27 @@ extern const char* platform_filter_template;
  */
 extern const char* native_filter_template;
 
+/**
+ * Template that enables the route cache reset filter in the chain.
+ * Should only be added when the route cache should be cleared on every request
+ * going through the filter chain between initial route resolution and the router
+ * filter's invocation on the request path. Typically only used for enabling
+ * direct responses to mutate headers which are then later used for routing.
+ */
+extern const char* route_cache_reset_filter_template;
+
+/**
+ * Template configuration used for creating "fake" remote clusters which enable
+ * local responses to be returned via direct response configurations.
+ */
+extern const char* fake_remote_cluster_template;
```

Comment:
It is a bit of a bummer that we can't hide/exclude this from the main target since we'll need to do the resolution at the configuration file :(

Reply:
What do you mean hide it from the main target?